dj_urls_panel/views.py

```
@@ -1,11 +1,71 @@
+            r'^192\.168\.',  # Private class C
+            r'^169\.254\.',  # Link-local (includes cloud metadata)
+            r'^::1$',  # IPv6 localhost
+            r'^fe80:',  # IPv6 link-local
+            r'^fc00:',  # IPv6 private
+        ]
+        
+        for pattern in blocked_patterns:
+            if re.match(pattern, hostname, re.IGNORECASE):
+                return False, f"Host '{hostname}' is blocked for security reasons (internal/private IP)"
+        
+        return True, None
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)